### ISSUE TITLE
http.HTTPStatus not cast to int in status field

### ISSUE THREAD

**Issue by arthurlm** [rycus86/prometheus_flask_exporter#54]
Hi

I have the following code that use `http` module for readability purposes.

```python
import http

from flask import Flask
from prometheus_flask_exporter import PrometheusMetrics

app = Flask("app")
metrics = PrometheusMetrics(app)

@app.route("/")
def index_json_no_content():
    return {}, http.HTTPStatus.NO_CONTENT

@app.route("/json")
def json_route():
    return {"foo": "bar"}

if __name__ == "__main__":
    app.run()
```

Flask perfectly understand `http.HTTPStatus`. It is an `enum.IntEnum`, so it is cast as `int` implicitly.
See bellow curl request / response:

```
> GET / HTTP/1.1
> Host: localhost:5000
> User-Agent: curl/7.58.0
> Accept: */*

< HTTP/1.0 204 NO CONTENT
< Content-Type: application/json
< Server: Werkzeug/1.0.1 Python/3.7.4
< Date: Fri, 05 Jun 2020 14:01:02 GMT
```

However, metrics generated does not cast `response.status_code` as `int`.
See:

- https://github.com/rycus86/prometheus_flask_exporter/blob/bed9760b59ebd2955f68d0d0b8223ddb3c368493/prometheus_flask_exporter/__init__.py#L343
- https://github.com/rycus86/prometheus_flask_exporter/blob/bed9760b59ebd2955f68d0d0b8223ddb3c368493/prometheus_flask_exporter/__init__.py#L348

So produce metrics are mixing `int` and `str` status_code.
See:

```
flask_http_request_duration_seconds_bucket{le="0.05",method="GET",path="/",status="HTTPStatus.NO_CONTENT"} 2.0
flask_http_request_duration_seconds_created{method="GET",path="/json",status="200"} 1.5913664307018697e+09
```

Is it possible to cast status_code as `int`, so flask user can use `http` module (or whatever class to implement `__int__`) ?
Or is there any other better way to let user be able to use `http` module and keep consistent results ?

**Comment by agjendem** [comment]
I ran into the same issue more than a year ago and solved it like this:

https://github.com/agjendem/prometheus_flask_exporter/commit/1d8e83491628a0507a873da86c669b4ecf7dcb1a

I didn't provide a PR (yet) as I failed at actually replicating the issue with a test case, and I'm not sure why. All tests I made with HttpStatus were happy without my modifications to the code. We did see (and fix) the problem in our actual applications though.
